src/web-animations-bonus-element-animatable.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//     You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function() {
+
+  if (document.createElement('div').animate([]).oncancel !== undefined) {
+    return;
+  }
+
+  var _now = undefined;
+  if (WEB_ANIMATIONS_TESTING) {
+    var now = function() { return webAnimations1.timeline.currentTime; };
+  } else {
+    var now = function() {
+      if (_now == undefined)
```

Comment:
This is copied from tick.js. Running the code it doesn't appear to always return the same time. If I understand correctly, because this is just defining the var "now", performance.now() or Date.now() are not executed at this point and are only executed when now() is called.

That said, I find this code a little confusing and possibly unnecessarily complex. Would the following be better?

```javascript
  if (WEB_ANIMATIONS_TESTING) {
    var now = function() { return webAnimations1.timeline.currentTime; };
  } else if (window.performance && performance.now) {
    var now = function() { return performance.now(); };
  } else {
    var now = function() { return Date.now(); };
  }
```